src/mdns/mdns_mdnssd.cpp

```diff
@@ -552,19 +665,33 @@ void PublisherMDnsSd::HandleRegisterHostResult(DNSServiceRef       aHostsConnect
     OTBR_UNUSED_VARIABLE(aHostsConnection);
     OTBR_UNUSED_VARIABLE(aFlags);
 
-    auto host =
-        std::find_if(mHosts.begin(), mHosts.end(), [&](const Host &aHost) { return aHost.mRecord == aHostRecord; });
-    assert(host != mHosts.end());
+    HostIterator host = FindPublishedHost(aHostRecord);
+    std::string  hostName;
+
+    VerifyOrExit(host != mHosts.end());
+    hostName = host->mName;
```

Comment:
seems not used.
may replace host->mName with hostName in below occurrences?

Reply:
thanks, resolved.